libs/workspace/mat-table/src/lib/components/mat-table-config-component/mat-table-config.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, EventEmitter, Output } from '@angular/core';
+import { MatTableDemoService } from '../../services/mat-table-demo.service';
+
+@Component({
+    templateUrl: './mat-table-config.component.html',
+    styleUrls: ['./mat-table-config.component.scss']
+})
+export class MatTableConfigComponent {
+    @Output() updatedComponentConfig = new EventEmitter();
+
+    componentConfiguration: any;
+
+    constructor(private matTableDemoService: MatTableDemoService) {}
+
+    toggleRowExpansion(): void {
+        this.componentConfiguration.enableRowExpansion = !this.componentConfiguration.enableRowExpansion;
```

Comment:
To be updated. Angular Material table needs a force update to make this work. Do you know how to externally force an update?